include/brickengine/components/physics_component.hpp

```diff
@@ -14,7 +16,11 @@ class PhysicsComponent : public ComponentImpl<PhysicsComponent> {
     double vx;
     double vy;
     bool gravity;
-    bool kinematic;
+    Kinematic kinematic;
+    // If the entity is a parent, only the direction will get flipped
+    // If the entity is a child, both the direction and position get flipped
+    bool flipX;
+    bool flipY;
```

Comment:
The direction and position are both on the same axis. So shouldn't this be called. `flip_direction` and `flip_position`

Reply:
The _flipX_ and _flipY_ vars both dictate whether to flip direction and position on their corresponding axis. For instance when you would want to fly or swim up you could set _flipY_ to true and you weapon would flip upwards with you. This is something that we will most certainly never ever use, but it does make the purpose clear and easier to use.

Reply:
Ah :+1: